### PR TITLE
add devcontainer setup

### PR DESCRIPTION
**Pull request by mabiede**
initial setup for development with Visual Studio Code and DevContainer

### CODE REVIEW

**Comment by mabiede** [comment]
If VSCode raises the following error from the OCaml platform package, degrade its version to 1.6.0.

`Couldn't start client OCaml Platform VS Code extension`

**Comment by jerben** [comment]
This PR makes it possible to run the tests in a dev container: https://github.com/oxidizing/sihl/pull/461

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "dev",
+    "postgres",
+    "mariadb"
+    // "adminer" // uncomment if adminer should run to see the database
+  ],
+  "workspaceFolder": "/workspace",
+  "remoteEnv": {
+    "DATABASE_URL": "postgres://admin:password@postgres:5432/dev",
+    "DATABASE_URL_TEST_MARIADB": "mariadb://admin:password@mariadb:5432/dev",
+    "DATABASE_URL_TEST_POSTGRES": "postgres://admin:password@postgres:5432/dev",
```
Comment (jerben) [inline .devcontainer/devcontainer.json:18]: @mabiede Let's use the full name to be consistent `DATABASE_URL_TEST_POSTGRESQL`

**Review by jerben — APPROVED** [review]
Looks good!

**Pull request merged**